### ISSUE TITLE
`reportAbstractUsage` conflicts with `reportMissingSuperCall`

### ISSUE THREAD

**Issue by DetachHead** [microsoft/pyright#7328]
```py
from abc import abstractmethod
from typing import override


class Foo:
    @abstractmethod
    def __enter__(self) -> None: ...

class Bar(Foo):
    @override
    def __enter__(self) -> None: # error: Method "__enter__" does not call the method of the same name in parent class (reportMissingSuperCall)
        ...
```

attempting to fix this triggers `reportAbstractUsage`:
```py
class Bar(Foo):
    @override
    def __enter__(self) -> None:
        super().__enter__() # error: Method "__enter__" cannot be called because it is abstract and unimplemented (reportAbstractUsage)
```
[playground](https://pyright-play.net/?strict=true&enableExperimentalFeatures=true&reportCallInDefaultInitializer=true&reportImplicitOverride=true&reportImplicitStringConcatenation=true&reportPropertyTypeMismatch=true&reportMissingSuperCall=true&reportShadowedImports=true&reportUninitializedInstanceVariable=true&reportUnnecessaryTypeIgnoreComment=true&reportUnusedCallResult=true&code=GYJw9gtgBAhgRgYygSwgBzCALrOBnLEGBLCAUywAswATAWAChRIosBPNZAOwHMV1MOMADcyIEMhplGMhggA2MPHigAxMGABcjKLqgABeASIlyVWjr1TgUAPq2yXLGPsAKPGXnAAlFAC0AHxQAHJgXGSaUAB0MbIKSioAQjAgrupg3toMegYiYhJSlrrWdg5OLrbunj7%2BQaHhkQDEUPmYRTnRsQyM8cpQyQBeaRqZ7fp54pLS2VZkNvaOziBuHl6%2BgSFhEe05eACuaGKu3lEL5cuVvs2tIEA)

**Comment by erictraut** [comment]
Pyright is working as designed here.

**Comment by dadwalharshal082-source** [comment]
please degin
